resources/groundwork/web.md

```diff
@@ -49,7 +49,64 @@ Have some technologies you enjoy using? See our [Contribution Guide](/CONTRIBUTI
+An abundance of external libraries and tooling available for these frameworks allows us to focus on application-specific logic instead of boilerplate code. | The abundance of tooling available can be initially overwhelming. |
+
+**Frontend**: React.js
+
+- Frameworks and libraries
+  - Component UI libraries: [`AntDesign`](https://ant.design/docs/react/introduce), [`MaterialUI`](https://material-ui.com/), [`SemanticUI`](https://semantic-ui.com/)
+  - [`React Router`](https://github.com/ReactTraining/react-router): allows us to build single-page applications with navigation
+  
+- Tooling
+  - For getting started, [`create-react-app`](https://create-react-app.dev/) produces React boilerplate setup with no configuration needed
+  - For debugging purposes, [`React Developer Tools`](https://chrome.google.com/webstore/detail/react-developer-tools/fmkadmapgofadopljbjfkapdkoienihi?hl=en#:~:text=React%20Developer%20Tools%20is%20a,%22%20and%20%22%E2%9A%9B%EF%B8%8F%20Profiler%22) is a Chrome extension that allows developers to inspect the component hierarchy
+  - For managing application state, [`Redux`](https://redux.js.org/) is a popular choice
+  - For build tools, [`Babel`](https://babeljs.io/docs/en/) transcompiles modern JavaScript and JSX into older versions for compatibility
+- Considerations
+  - Pro: React's popularity in the community means that there is no shortage of learning resources
+  - Con: React is less opinionated than other front-end frameworks like Vue or Angular and can become difficult to structurally maintain in the long run. React also uses a lot of "magic" and beginners diving deep into it without learning JavaScript fundamentals beforehand often find themselves struggling later
+
+- References
+  - [React documentation](https://reactjs.org/)
+  - [JavaScript resources](/resources/languages.md#javascript)
+  - [TypeScript resources](/resources/languages.md#typescript)
+
+**Backend**: Node.js
+
+- Frameworks and libraries
+  - [`Express.js`](https://expressjs.com/): a minimal web application framework used for building REST APIs
+  - [`Socket.io`](https://socket.io/): allows us to manage real-time event-based communication using web sockets
+- Tooling
+  - For productivity, [`nodemon`](https://nodemon.io/) automatically restarts the application upon changes detected
+- Considerations
+  - Pro: Node.js applications scale up well and are known for being performant. Like React, it is also backed by a large community and hence, learning resources are always available
+  - Con: Being a single-threaded environment, Node.js is not suitable for CPU-intensive applications (e.g. audio/video processing)
+- References
+  - [Node.js documentation](https://nodejs.org/en/)
+
+**Datastore**: MongoDB
```

Comment:
Bonus: mongo seems like a popular choice - it would be nice if we could link to examples, such as:

https://sourcegraph.com/search?q=r:ubclaunchpad/*+fork:no+mongo+-file:%28json%7Clock%29&patternType=literal

It's not a very refined query and frankly it's probably not very useful to everyone, but personally I find seeing how other projects use things very helpful :)

Reply:
We have provided the example of food-doods (https://github.com/ubclaunchpad/food-doods) in line #107 -- that has a clear demonstration of how mongo (with mongoose) can be used in a node.js + express project